draft-ietf-tls-tls13.md

```diff
@@ -2364,35 +2380,49 @@ RSASSA-PKCS1-v1_5 algorithms
   with the corresponding hash algorithm as defined in {{!SHS=DOI.10.6028/NIST.FIPS.180-4}}. These values
   refer solely to signatures which appear in certificates (see
   {{server-certificate-selection}}) and are not defined for use in signed
-  TLS handshake messages.
+  TLS handshake messages, although they MAY appear in "signature_algorithms"
+  and "signature_algorithms_cert" for backward compatibility with TLS 1.2,
 
 ECDSA algorithms
 : Indicates a signature algorithm using ECDSA {{ECDSA}}, the corresponding
   curve as defined in ANSI X9.62 {{X962}} and FIPS 186-4 {{DSS}}, and the
   corresponding hash algorithm as defined in {{!SHS}}. The signature is
   represented as a DER-encoded {{X690}} ECDSA-Sig-Value structure.
 
-RSASSA-PSS algorithms
+RSASSA-PSS RSAE algorithms
 : Indicates a signature algorithm using RSASSA-PSS {{RFC8017}} with mask
   generation function 1. The
   digest used in the mask generation function and the digest being signed are
-  both the corresponding hash algorithm as defined in {{!SHS}}. When used in
-  signed TLS handshake messages, the length of the salt MUST be equal to the
-  length of the digest output.  This codepoint is new in this document and is also
-  defined for use with TLS 1.2.
+  both the corresponding hash algorithm as defined in {{!SHS}}.
+  The length of the salt MUST be equal to the length of the digest
+  algorithm. If the public key is carried
+  in an X.509 certificate, it MUST use the rsaEncryption OID.
 
 EdDSA algorithms
 : Indicates a signature algorithm using EdDSA as defined in
   {{RFC8032}} or its successors. Note that these correspond to the
   "PureEdDSA" algorithms and not the "prehash" variants.
 
+RSASSA-PSS PSS algorithms
+: Indicates a signature algorithm using RSASSA-PSS {{RFC8017}} with mask
+  generation function 1. The
+  digest used in the mask generation function and the digest being signed are
+  both the corresponding hash algorithm as defined in {{!SHS}}.
+  The length of the salt MUST be equal to the length of the digest
+  algorithm. If the public key is carried in an X.509 certificate,
```

Comment:
So now we are enforcing salt length restrictions on signatures in X509 certificates (both here and above)?